po/ru.po

```diff
@@ -56,6 +56,8 @@ msgstr "День 1: Утро"
 msgid "Welcome"
 msgstr "Добро пожаловать"
 
+# There should be two separate keys (or additional context): Hello Rust and Hello, World.
```

Comment:
Thanks to @dyoo, it is actually possible to embed translation comments in the Markdown files. See https://github.com/google/mdbook-i18n-helpers/pull/177 where he adds the documentation to our tooling.

I don't think we support setting the [translation context](https://www.gnu.org/software/gettext/manual/html_node/Contexts.html) yet. @dyoo, do you know?

@1dimir, if you have suggestions for changes to the English text (which is far from perfect!), then feel free to submit PRs.

Reply:
Yeah, we haven't added that, though I suspect it can follow a similar pattern to how we're doing comments.

So probably a sketch of supporting this would be:

* Define another comment directive to let authors specify the ctxt, similar to what we're doing [here](https://github.com/google/mdbook-i18n-helpers/blob/main/i18n-helpers/src/directives.rs#L17-L35)

* Collect and thread that state when we're [gathering groups](https://github.com/google/mdbook-i18n-helpers/blob/main/i18n-helpers/src/lib.rs#L154-L171) for translation.

* Finally, pass the `ctxt` [here](https://github.com/google/mdbook-i18n-helpers/blob/main/i18n-helpers/src/xgettext.rs#L43) and call the builder method for [setting the `ctxt`](
https://docs.rs/polib/latest/polib/message/struct.MessageBuilder.html#method.with_msgctxt).

So I think this would be fairly straightforward extension.

Reply:
Thanks @dyoo, I took the above and used it to create https://github.com/google/mdbook-i18n-helpers/issues/184.